examples/blockdiagperm.cc

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include <ATen/ATen.h>
+
+#include "tc/aten/aten_compiler.h"
+#include "tc/autotuner/genetic_autotuner_aten.h"
+#include "tc/core/cuda/cuda.h"
+#include "tc/core/cuda/cuda_tc_executor.h"
+#include "tc/core/flags.h"
+#include "tc/core/mapping_options.h"
+
+TEST(BlockDiagPerm, SimpleAutotune) {
+  // 1. Define and setup the TC compilation unit with CUDA memory
+  // management backed by ATen tensors.
+  std::string tc = R"TC(
+# This TC requires de-linearization + dependence analysis
+# def blockdiagperm2dfused(float(B, K, NBYK) I, float(K, NBYK, NBYK) W, float(K, NBYK) Idx)
+#     -> (O1) {
+#     O1(b, k, nbyk1) +=!  I(b, k, r_nbyk0) * W(k, r_nbyk0, nbyk1)
+#     O2(b, k, nbyk)   =  O1(b, Idx(k, nbyk) / NBYK, Idx(k, nbyk) % NBYK)
+# }
```

Comment:
Please explain somewhere (say, the commit message) why you are introducing
commented out code.  I can guess, but it's better to explain explicitly to avoid
someone eagerly removing commented out code later.

About this particular TC, are you not using it because it doesn't work somehow?
If so, why does it not work?
I don't see why you would need de-linearization.
I also don't see why you mention dependence analysis specifically.
We're applying dependence analysis already, aren't we?